pointcloud/basic_octree_test.go

```diff
@@ -598,3 +600,44 @@ func testPCDToBasicOctree(t *testing.T, artifactPath string) {
 
 	validateBasicOctree(t, basicOct, basicOct.center, basicOct.sideLength)
 }
+
+func TestCachedMaxProbability(t *testing.T) {
+	center := r3.Vector{X: 0, Y: 0, Z: 0}
+	side := 2.0
+
+	octree, err := createNewOctree(center, side)
+	test.That(t, err, test.ShouldBeNil)
+
+	pointsAndData := []PointAndData{
+		{P: r3.Vector{X: 0, Y: 0, Z: 0}, D: NewValueData(2)},
+		{P: r3.Vector{X: .5, Y: 0, Z: 0}, D: NewValueData(3)},
+		{P: r3.Vector{X: .5, Y: 0, Z: .5}, D: NewValueData(10)},
+		{P: r3.Vector{X: .5, Y: .5, Z: 0}, D: NewValueData(1)},
+		{P: r3.Vector{X: .55, Y: .55, Z: 0}, D: NewValueData(4)},
+		{P: r3.Vector{X: -.55, Y: -.55, Z: 0}, D: NewValueData(5)},
+		{P: r3.Vector{X: .755, Y: .755, Z: 0}, D: NewValueData(6)},
+	}
+
+	err = addPoints(octree, pointsAndData)
+	test.That(t, err, test.ShouldBeNil)
+
+	validateBasicOctree(t, octree, octree.center, octree.sideLength)
+
+	// test that we can query the tree for max prob
+	mp, err := getMaxProb(octree, r3.Vector{X: 0, Y: 0, Z: 0})
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, mp, test.ShouldEqual, 0.02)
+
+	mp, err = getMaxProb(octree, r3.Vector{X: .755, Y: .755, Z: 0})
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, mp, test.ShouldEqual, 0.06)
```

Comment:
So far it looks like all the things you test miss the recursive case, so please add a test that catches that the recursion is propapagating the correct value up the tree

Reply:
spoke offline about how the recursive test case is fulfilled.